### PR TITLE
Damp and mould reports view page

### PR DESCRIPTION
**Pull request by LBHCallumM**
## Ticket
[https://hackney.atlassian.net/browse/MR-868?atlOrigin=eyJpIjoiNTcyMjYwODM0NzllNGRlZGJjN2NlYjY4MjBmYTQzMzUiLCJwIjoiaiJ9](https://hackney.atlassian.net/browse/MR-868?atlOrigin=eyJpIjoiNTcyMjYwODM0NzllNGRlZGJjN2NlYjY4MjBmYTQzMzUiLCJwIjoiaiJ9)

## Summary of Changes
Add damp and mould report view, consisting of an 'all reports' page and a 'reports per property' page

- The 'all reports' page uses a pagination approach, similar to MMH search
- The 'reports per property' page uses a 'show more' button. This is the approach for showing workOrders against a property in RH
- Added 'Damp and mould reports' header link for authorized users. 
- Updated the header styles to be more responsive. I needed to allow the header links to wrap onto a second line.

Pages:
- `/damp-and-mould-reports`
- `/damp-and-mould-reports/{propertyReference}`

### Authorization
The pages are authorised by a new `DAMP_AND_MOULD_REPORT_MANAGER` role.

### View all reports page

image

image

### Reports for property page

image

### Updated header

image

### CODE REVIEW

File: cypress/integration/damp-and-mould/damp-and-mould-reports-property.spec.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
+/// <reference types="cypress" />
+
+import 'cypress-audit/commands'
+import reportsFixture from '../../fixtures/dampAndMouldReports/reports'
+
+const PROPERTY_REFERENCE = '001234567788'
+
+describe('Damp and mould reports property page page', () => {
```
Comment (adamtry) [inline cypress/integration/damp-and-mould/damp-and-mould-reports-property.spec.js:8]: page repeated

File: src/components/DampAndMouldReports/components/DampAndMouldReportsPropertyReportsView.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        Showing {reports.length ?? 0} of {reportsApiResponse?.totalCount ?? 0}{' '}
+        reports
+      </p>
+      {isLoading ? (
+        <Spinner />
+      ) : (
+        <>
+          <DampAndMouldReportTable
+            reports={reports}
+            showOtherReportsColumn={false}
```
Comment (adamtry) [inline src/components/DampAndMouldReports/components/DampAndMouldReportsPropertyReportsView.js:75]: You could consider using a single boolean for both column flags to make the intent more clear, e.g. `showAdditionalPropertyInfo`.

Comment (LBHCallumM) [reply src/components/DampAndMouldReports/components/DampAndMouldReportsPropertyReportsView.js:75]: Yeah, if you think its clearer

**Review by adamtry — APPROVED** [review]
Looks good

File: src/components/DampAndMouldReports/components/DampAndMouldReportsPropertyReportsView.js (hunk 2 above)
Comment (adamtry) [reply src/components/DampAndMouldReports/components/DampAndMouldReportsPropertyReportsView.js:75]: It took me a bit to figure out why these attributes existed

File: src/components/DampAndMouldReports/components/ReportFrequencyBadge.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  return (
+    <div style={{ marginTop: '5px', marginLeft: '-5px' }}>
+      <span
+        className={cx([
+          'damp-and-mould-report-frequency-badge ',
+          numberOfReports === 0 ? 'new-report' : 'recurring-report',
+        ])}
+      >
+        {numberOfReports === 0
+          ? 'New report'
```
Comment (adamtry) [inline src/components/DampAndMouldReports/components/ReportFrequencyBadge.js:13]: Should it display 'New report' in this case?

I checked and it displays this even if all the reports are many months in the past

**Pull request merged**